rasa/core/policies/ensemble.py

```diff
@@ -578,16 +578,29 @@ def _pick_best_policy(
 
         form_confidence = None
         form_policy_name = None
-        # End-to-end predictions overrule all other predictions.
-        use_only_end_to_end = any(
+        # different type of predictions have different priorities
+        # No user predictions overrule all other predictions.
+        is_no_user_prediction = any(
+            prediction.is_no_user_prediction for prediction in predictions.values()
+        )
+        # End-to-end predictions overrule all other predictions based on user input.
+        is_end_to_end_prediction = any(
             prediction.is_end_to_end_prediction for prediction in predictions.values()
         )
-        policy_events = []
 
+        policy_events = []
         for policy_name, prediction in predictions.items():
             policy_events += prediction.events
 
-            if prediction.is_end_to_end_prediction != use_only_end_to_end:
+            # No user predictions overrule all other predictions.
+            if prediction.is_no_user_prediction != is_no_user_prediction:
```

Comment:
seems a tiny bit easier to understand for me, what do you think? 
```suggestion
            if is_no_user_prediction and not prediction.is_no_user_prediction:
```

Reply:
it doesn't seem to me easier to understand, took me some time to grasp that it would be the same thing